src/net_processing.cpp

```diff
@@ -2021,8 +2021,8 @@ static void ProcessHeadersMessage(CNode& pfrom, CConnman& connman, ChainstateMan
         if (!pfrom.fDisconnect && pfrom.IsFullOutboundConn() && nodestate->pindexBestKnownBlock != nullptr) {
             // If this is an outbound full-relay peer, check to see if we should protect
             // it from the bad/lagging chain logic.
-            // Note that block-relay-only peers are already implicitly protected, so we
-            // only consider setting m_protect for the full-relay peers.
+            // Note that outbound block-relay peers are excluded from this protection and
+            // thus passive of eviction by bad/lagging chain logic.
```

Comment:
How about:

```
// Note that outbound block-relay peers are excluded from this protection, and thus always subject to eviction under the bad/lagging chain logic.
```

Reply:
Also, you may want to consider adding a comment that explains how `m_protect` is used by stale-tip-based eviction as well.  In particular, the stale-tip -> extra full outbound peer logic could cause an existing full outbound to be evicted, but peers with `m_protect` set will not be chosen for eviction.

Reply:
Thanks for the suggestion. Also added a commit on scope of m_protect w.r.t to both eviction logics.